src/Forms/NumericField.php

```diff
@@ -132,71 +205,105 @@ public function getSchemaValidation()
+        return $this->cast($this->value);
+    }
 
-        if (!$this->isNumeric()) {
-            return 0;
+    /**
+     * Gets the current locale this field is set to.
+     *
+     * @return string
+     */
+    public function getLocale()
+    {
+        if ($this->locale) {
+            return $this->locale;
         }
 
-        $locale = new Zend_Locale($this->getLocale());
+        return i18n::get_locale();
+    }
 
-        $number = Zend_Locale_Format::getNumber(
-            $this->clean($this->value),
-            array('locale' => $locale)
-        );
+    /**
+     * Override the locale for this field.
+     *
+     * @param string $locale
+     *
+     * @return $this
+     */
+    public function setLocale($locale)
+    {
+        $this->locale = $locale;
 
-        return $number;
+        return $this;
     }
 
     /**
-     * Creates a read-only version of the field.
+     * Determine if we should use html5 number input
      *
-     * @return NumericField_Readonly
+     * @return bool
      */
-    public function performReadonlyTransformation()
+    public function getHTML5()
     {
-        $field = new NumericField_Readonly(
-            $this->name,
-            $this->title,
-            $this->value
-        );
-
-        $field->setForm($this->form);
+        return $this->html5;
+    }
 
-        return $field;
+    /**
+     * Set whether this field should use html5 number input type.
+     * Note: If setting to true this will disable all number localisation.
+     *
+     * @param bool $html5
+     * @return $this
+     */
+    public function setHTML5($html5)
+    {
+        $this->html5 = $html5;
+        return $this;
     }
 
     /**
-     * Gets the current locale this field is set to.
+     * Step attribute for html5. E.g. '0.01' to enable two decimal places.
+     * Ignored if html5 isn't enabled.
      *
      * @return string
      */
-    public function getLocale()
+    public function getStep()
     {
-        if ($this->locale) {
-            return $this->locale;
+        $scale = $this->getScale();
+        if ($scale === null) {
+            return 'any';
         }
-
-        return i18n::get_locale();
+        if ($scale === 0) {
+            return '1';
+        }
+        return '0.'.str_repeat('0', $scale - 1).'1';
     }
 
     /**
-     * Override the locale for this field.
+     * Get number of digits to show to the right of the decimal point.
+     * 0 for integer, any number for floating point, or null to flexible
      *
-     * @param string $locale
+     * @return int|null
+     */
+    public function getScale()
+    {
+        return $this->scale;
+    }
+
+    /**
+     * Get number of digits to show to the right of the decimal point.
+     * 0 for integer, any number for floating point, or null to flexible
      *
+     * @param int|null $scale
      * @return $this
      */
-    public function setLocale($locale)
+    public function setScale($scale)
```

Comment:
Actually, both left/right digits are "significant" - quoting https://en.wikipedia.org/wiki/Significant_figures#Rounding_and_decimal_places

> Trailing zeros in a number containing a decimal point are significant. For example, 12.2300 has six significant figures: 1, 2, 2, 3, 0 and 0

But yeah, you're correct that "significant digits" is the wrong term here. I'm used to "decimal places", but since you've documented it well, I'm also happy with "scale"